planner/cascades/implementation_rules.go

```diff
@@ -221,3 +224,22 @@ func (r *ImplSort) OnImplement(expr *memo.GroupExpr, reqProp *property.PhysicalP
 	)
 	return impl.NewSortImpl(ps), nil
 }
+
+// ImplHashAgg is the implementation rule which implements LogicalAggregation
+// to PhysicalHashAgg.
+type ImplHashAgg struct {
+}
+
+// Match implements ImplementationRule Match interface.
+func (r *ImplHashAgg) Match(expr *memo.GroupExpr, prop *property.PhysicalProperty) (matched bool) {
+	// TODO: deal with the hints when we have implemented StreamAgg.
+	return prop.IsEmpty()
```

Comment:
Would this rule implement TiKVHashAgg as well later? if not, shall we check engine type here?

Reply:
Yes, I plan to implement TiKVHashAgg in this rule, it may reduce the function call of `Match`.